akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -89,3 +89,11 @@ class DnsExt(system: ExtendedActorSystem) extends IO.Extension {
 
   def getResolver: ActorRef = manager
 }
+
+object IpVersionSelector {
+  def getInetAddress(ipv4: Option[Inet4Address], ipv6: Option[Inet6Address]): Option[InetAddress] =
+    sys.props.get("java.net.preferIPv6Addresses") match {
```

Comment:
Looks costly to the system property lookup each time. Putting it in a val makes testing difficult. Question is if this really needs a tests. Pretty trivial logic here.